packages/frontend/src/theory/types.ts

```diff
@@ -6,6 +6,66 @@ import type { DiagramAnalysisComponent, ModelAnalysisComponent } from "../analys
 import { uniqueIndexArray } from "../util/indexing";
 import type { ArrowStyle } from "../visualization";
 
+/** Data about privileged migrations a frontend theory can have.
+ */
+export class MapData {
```

Comment:
I'm not completely sure but my instinct is that this whole type should be define in Rust and exposed via Wasm. We can discuss the design more.

Reply:
Thanks I like this idea! I wasn't sure where the data structure belonged in the code itself but I defined it just above where it gets used (in `model.rs`) to start with.